engine/internal/runtime/manager.go

```diff
@@ -290,16 +312,38 @@ func (m *Manager) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result,
 		}
 		m.markRuntimeReady(sts.Name, modelID, addr)
 		log.Info("Runtime is ready")
+	} else {
+		var podList corev1.PodList
+		if err := m.k8sClient.List(ctx, &podList,
+			client.InNamespace(sts.Namespace),
+			client.MatchingLabels{"app.kubernetes.io/instance": sts.Name},
+		); err != nil {
+			log.Error(err, "Failed to list pods")
+			return ctrl.Result{}, err
+		}
+		for _, pod := range podList.Items {
+			if pod.Labels["controller-revision-hash"] != sts.Status.CurrentRevision {
+				continue
+			}
+			for _, cond := range pod.Status.Conditions {
+				if cond.Type == corev1.PodScheduled {
```

Comment:
No strong opinion, but we can combine these two `if`s into one.

Reply:
The first `if` is used for an early break.